### PR TITLE
Features/26

### PR DESCRIPTION
**Pull request by dortegau**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions
* Remember to set **idealista:develop** as base branch;

### Description of the Change

Deploying to [Docker Hub](https://cloud.docker.com/u/idealista/repository/docker/idealista/zookeeper) using Ansible Playbooks (**only for testing purposes**)

### Benefits

Some unit tests using Molecule and Docker as driver ([kafka-role](https://github.com/idealista/kafka-role), [solrcloud-role](https://github.com/idealista/solrcloud-role)) will be speed up because ZooKeeper role execution is no longer needed.

### Possible Drawbacks

This image **should be used only for testing purposes**. Deployment assumes a unique rolling tag called "latest" and major changes could broke services depending on this image.

The container should be run with privileges:

`docker run --privileged -v /sys/fs/cgroup:/sys/fs/cgroup -it idealista/zookeeper:latest`

To understand differences between rolling tags and immutable tags: https://docs.bitnami.com/containers/how-to/understand-rolling-tags-containers/

### Applicable Issues

#26

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,4 +10,4 @@ tests/__pycache__
 *.iml
 .idea
 Pipfile
-Pipfile.lock
+Pipfile.lock
```
Comment (jnogol) [inline .gitignore]: `Pipfile*`, maybe?

File: .travis.yml [hunk 2]
```diff
@@ -1,15 +1,26 @@
...
+  - secure: j7g0vouk54s0Z1fGwuF6iTk0ZwIzdLGB5isT4dtYYpyI+N+jdu1OrXX/gazZujsw7eNl5dICj0n0mbfPV6lrxegJfB4xdm6uvF0fODj5dbg8kdywsI8lxpd7nbAyWAUClAXw8ij+5ItIlIkHezge846olneU8JI31RxijDo1SvMdkd1PWlrbW0EBv7iAAA2Ex7HIM9MxBYFZKpxwFFcjawiJFyNLunuhPGsyHdMc51uLfq/eIR0hRaZxTqyR2nd9r9dG/4xlaxAuVycD4tOyp4poIri+vkOS5Ff3dtUCv2KWIbiNAG8fcEXjwkmHJrwYTwv9eEtmLZwWgTAVFOMV2DEsR+y6EeZU3g8UT8TZc6pbylvhyihgLWp6UfVJjyqzYFXYhu5JO+QOUIrv/I3BnKWmEt8xnwIXhydNQJPnhnGDsde+UoUk5ZMUhBimOAxic2AonIhMHJk6w2nWzZIwHeBjsIAfsf89zy37TIgL2KrqTztOHZ8B6OgB9TdF756R2zBa8wsUdxav3XluxMLy74jTzHQdZWKljVLyeSMdjwO6hm7LVWvYp49e8IWhMcWO5rdTrhsYkXClTRZdI8BitoHGAiN/0eEIH2uO7lmXQrl1SqYtMbtYrO/7z+Jldt61bLvmn9mVoweK0YQYZMxSaY8LCnOBNiksIryGrNeJ2Zk=
+
+script:
+- pipenv run molecule test
 notifications:
   webhooks: https://galaxy.ansible.com/api/v1/notifications/
+deploy:
+  provider: script
+  script:
+  - ANSIBLE_ROLES_PATH=../ pipenv run ansible-playbook dockerhub/deploy.yml --extra-vars "docker_hub_email=${DOCKER_EMAIL} docker_hub_username=${DOCKER_USERNAME} docker_hub_password=${DOCKER_PASSWORD}"
```
Comment (jnogol) [inline .travis.yml]: nice touch, setting the email via variable

File: dockerhub/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        zookeeper idealista/zookeeper
+
+    - name: Log into Docker Hub
+      docker_login:
+        email: "{{ docker_hub_email }}"
+        username: "{{ docker_hub_username }}"
+        password: "{{ docker_hub_password }}"
+    
+    - name: Tag and push to Docker Hub
+      command: docker push idealista/zookeeper:latest
```
Comment (jnogol) [inline dockerhub/deploy.yml]: always `latest`? No tagging?

File: dockerhub/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,9 @@
+FROM idealista/jdk:8u181-stretch-openjdk-headless
+LABEL maintainer="idealista <labs@idealista.com>"
+
+RUN apt-get update \
+    && apt-get install -y sudo systemd systemd-sysv
+
+EXPOSE 2181 2888 3888
```
Comment (jnogol) [inline dockerhub/Dockerfile]: are those needed for testing purposes?

**Comment by dortegau** [comment]
> lint problems in CI :(

fixed!

File: .gitignore (hunk 1 above)
Comment (dortegau) [reply .gitignore]: Fixed!

**Pull request merged**